pastastore/util.py

```diff
@@ -399,3 +400,232 @@ def copy_database(conn1, conn2, libraries: Optional[List[str]] = None,
+                            check_gain_passed = pd.NA
+                        elif np.isnan(gain_std):
+                            check_gain_passed = pd.NA
+                        else:
+                            check_gain_passed = np.abs(gain) > 2 * gain_std
+                        checks.loc[f"gain > 2*std: {sm_name}-{iw:02g}"] = (
+                            gain, 2 * gain_std, "m/(Mm3/yr)", check_gain_passed)
+                else:
+                    gain = ml.parameters.loc[f"{sm_name}_A", "optimal"]
+                    gain_std = ml.parameters.loc[f"{sm_name}_A", "stderr"]
+                    if gain_std is None:
+                        gain_std = np.nan
+                        check_gain_passed = pd.NA
+                    elif np.isnan(gain_std):
+                        check_gain_passed = pd.NA
+                    else:
+                        check_gain_passed = np.abs(gain) > 2 * gain_std
+                    check_gain_passed = np.abs(gain) > 2 * gain_std
+                    checks.loc[f"gain > 2*std: {sm_name}"] = (
+                        gain, 2 * gain_std, "m/(Mm3/yr)", check_gain_passed)
+
+        # Check 5 - Parameter Bounds
+        if check5:
+            upper, lower = ml._check_parameters_bounds()
+            for param in ml.parameters.index:
+                bounds = (ml.parameters.loc[param, "pmin"],
+                          ml.parameters.loc[param, "pmax"])
+                b = ~(upper.loc[param] or lower.loc[param])
+
+                checks.loc[f"Parameter bounds: {param}", :] = (
+                    ml.parameters.loc[param, "optimal"], bounds, "_", b)
+
+        if checks['check_passed'].all():
+            check_passed_models.append(ml.name)
+
+        if path:
+            checks.to_csv(f"{path}/checks_{ml.name}.csv",
+                          na_rep="NaN")
+
+    return check_passed_models
+
+
+def frontiers_select(pstore, modelnames):
+    """Select the best model structure based on the AIC as proposed
+    by Brakenhoff et al. 2022 [bra_2022]_.
+
+    Parameters
+    ----------
+    pstore : pastastore object
+    modelnames : list
+        list of models (that pass reliability criteria)
+
+    Returns
+    -------
+    pandas DataFrame
+        DataFrame with oseries locations, models that pass the check
+        per location, AIC per model and model with the lowest AIC.
+
+    References
+    ----------
+    .. [bra_2022] Brakenhoff, D.A., Vonk M.A., Collenteur, R.A.,
+    van Baar, M., Bakker, M.: Application of Time Series Analysis
+    to Estimate Drawdown From Multiple Well Fields.
+    Front. Earth Sci., 14 June 2022 doi:10.3389/feart.2022.907609
+    """
+
+    # Dataframe of models with corresponding oseries
+    df = pstore.get_model_timeseries_names(
+        modelnames, progressbar=False).loc[:, ['oseries']]
+    # AIC of models
+    aic = pstore.get_statistics(['aic'], modelnames)
+    # Group models per location and obtain the AIC identify model with lowest AIC per location
```

Comment:
This feels like it could be simpler? 

```python
df.join(aic).groupby("oseries").idxmin()  # model name with lowest AIC
df.join(aic).groupby("oseries").min()  # lowest AIC
```

(Edit groupby oseries, not model names)

Reply:
Could probably be simpler but I like it the end table to be:

| oseries    | Model Names                                                              | AIC                                                                         | Min_AIC        |
|:-----------|:-------------------------------------------------------------------------|:----------------------------------------------------------------------------|:---------------|
| B27A0431   | ['B27A0431_000', 'B27A0431_020']                                         | [-51731.37683039733 -51714.6343188431]                                      | B27A0431_000   |
| B32E0125   | ['B32E0125_000', 'B32E0125_020']                                         | [-46066.864554639826 -46061.45786205343]                                    | B32E0125_000   |

This way we can easily plot all reliable models per location and compare the difference in AIC.

Maybe also a column with the difference in AIC?

Reply:
I kind of like having both levels available in this function, I think.